### PR TITLE
Place upper bounds on inspec/rubcop

### PR DESCRIPTION
**Pull request by decentral1se**
Refs:
  * https://github.com/ansible/molecule/pull/2045
  * https://github.com/ansible/molecule/issues/2042
  * https://github.com/inspec/inspec/blob/master/CHANGELOG.md
  * https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md

I'm OK with having inspec/rubocop users having to come and ask for new major version tool support for the immediate short term. We can bake in a forward looking major version build into the CI more mid-term.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -608,6 +608,7 @@ install:
   - >
     ! grep 'ERROR: ' ~/.tox-venv-install.log .tox/*/log/*.log
 before_script:
-  - gem install inspec rubocop
+  - gem install inspec -v '~> 3'
```
Comment (decentral1se) [inline .travis.yml]: `gem` does not support inlining this from my testing. 

If anyone knows how to do it better, I am all ears.

`~>` means, bound before the next major version.

Comment (seandst) [reply .travis.yml]: One suggestion would be to use a Gemfile and bundler, which (imo) is a lot of extra complexity just to get this back down to one line. I think doing it on two separate lines is fine.

Comment (seandst) [reply .travis.yml]: Also, a little bit of interactive ruby to demo how the ~> comparator works for those that may not be familiar with it. In this case, `~> 3` expands to `>= 3, < 4`:

```rb
$ irb
irb(main):001:0> depcheck = Gem::Dependency.new('inspec', '~> 3')
=> <Gem::Dependency type=:runtime name="inspec" requirements="~> 3">
irb(main):002:0> depcheck.match?('inspec', '2.99')
=> false
irb(main):003:0> depcheck.match?('inspec', '3.0')
=> true
irb(main):004:0> depcheck.match?('inspec', '3.99')
=> true
irb(main):005:0> depcheck.match?('inspec', '4.0')
=> false
```

**Comment by decentral1se** [comment]
Can someone with commit bit +1 this as well? Thanks.

**Comment by decentral1se** [comment]
Looks good then @webknjaz?

**Comment by webknjaz** [comment]
I don't know enough about Ruby to make a judgment here. I'll leave it to somebody better knowledgeable.

**Comment by decentral1se** [comment]
@webknjaz, not many are around with the commit bit to get this merged, so that is why I am explicitly asking you to approve this. Given that @seandst has approved this and seems to have more Ruby knowledge than us, let's just go with it. If there are issues, I will follow up prompty, as usual.

**Pull request merged**